src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -330,22 +377,45 @@ class npc_snobold_vassal : public CreatureScript
                 switch (pointId)
                 {
                     case 0:
-                        if (_targetDied)
+                    {
+                        Unit* victim = ObjectAccessor::GetUnit(*me, _targetGUID);
+                        if (victim && !victim->IsAlive())
                             me->DespawnOrUnsummon();
                         break;
+                    }
                     default:
                         break;
                 }
             }
 
             void JustDied(Unit* /*killer*/) override
             {
-                if (Unit* target = ObjectAccessor::GetPlayer(*me, _targetGUID))
-                    if (target->IsAlive())
-                        target->RemoveAurasDueToSpell(SPELL_SNOBOLLED);
+                Unit* victim = ObjectAccessor::GetUnit(*me, _targetGUID);
+                if (victim && victim->IsAlive())
+                    victim->RemoveAurasDueToSpell(SPELL_RIDE_PLAYER);
+
                 _instance->SetData(DATA_SNOBOLD_COUNT, DECREASE);
             }
 
+            void SetGUID(ObjectGuid guid, int32 id) override
+            {
+                switch (id)
+                {
+                    case DATA_NEW_TARGET:
+                        if (Unit* target = ObjectAccessor::GetPlayer(*me, guid))
+                        {
+                            me->getThreatManager().resetAllAggro();
+                            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE);
+                            _targetGUID = target->GetGUID();
+                            me->TauntApply(target);
```

Comment:
On my live server, sometimes snobolds dont attack melee the right player (he try attack anothers)
so i added ```me->getThreatManager().addThreat(victim, 1000.0f);``` here
And he is immune to taunt (need add extra flag em DB)